modules/age.nix

```diff
@@ -17,15 +17,28 @@ let
   identities = builtins.concatStringsSep " " (map (path: "-i ${path}") cfg.sshKeyPaths);
   installSecret = secretType: ''
     echo "decrypting ${secretType.file} to ${secretType.path}..."
-    TMP_FILE="${secretType.path}.tmp"
-    mkdir -p $(dirname ${secretType.path})
+
+    if [[ "$NIXOS_ACTION" == "dry-activate" ]]; then
+      OUT_DIR="$(mktemp -d)"
+    else
+      OUT_DIR="$(dirname ${secretType.path})"
+      mkdir -p "$OUT_DIR"
+    fi
+
+    TMP_FILE="$OUT_DIR/tmp"
     (
       umask u=r,g=,o=
       LANG=${config.i18n.defaultLocale} ${ageBin} --decrypt ${identities} -o "$TMP_FILE" "${secretType.file}"
     )
     chmod ${secretType.mode} "$TMP_FILE"
     chown ${secretType.owner}:${secretType.group} "$TMP_FILE"
-    mv -f "$TMP_FILE" '${secretType.path}'
+
+    if [[ "$NIXOS_ACTION" == "dry-activate" ]]; then
+      fcho "[agenix] dry-run, not moving decrypted secret"
```

Comment:
fcho?

Reply:
Sorry, I had to leave in a rush and wasn't able to test this properly yet. Just wanted to share the progress I already made as draft.